homeassistant/components/tplink_omada/device_tracker.py

```diff
@@ -0,0 +1,206 @@
+        is_connected_fn=lambda client: client.is_active if client else False,
+    )
+]
+
+
+class OmadaBaseTrackerEntity(CoordinatorEntity[OmadaClientsCoordinator]):
+    """Base entity for tracking a client connected to the Omada network."""
+
+    entity_description: OmadaDeviceTrackerEntityDescription
+    _client_details: OmadaWirelessClient | None = None
+
+    def __init__(
+        self,
+        site_id: str,
+        client_id: str,
+        display_name: str,
+        coordinator: OmadaClientsCoordinator,
+        entity_description: OmadaDeviceTrackerEntityDescription,
+    ) -> None:
+        """Initialize the scanner."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+        self._site_id = site_id
+        self._client_id = client_id
+        self._attr_name = display_name
+
+    @property
+    def source_type(self) -> SourceType:
+        """Return the source type of the device."""
+        return SourceType.ROUTER
+
+    def _do_update(self) -> None:
+        self._client_details = self.entity_description.coordinator_update_func(
+            self.coordinator, self._client_id
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """When entity is added to hass."""
+        await super().async_added_to_hass()
+        self._do_update()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._do_update()
+        self.async_write_ha_state()
+
+
+class OmadaClientScannerEntity(OmadaBaseTrackerEntity, ScannerEntity):
+    """Entity for a client connected to the Omada network."""
+
+    @property
+    def ip_address(self) -> str | None:
+        """Return the primary ip address of the device."""
+        return self.entity_description.ip_address_fn(self._client_details)
+
+    @property
+    def mac_address(self) -> str | None:
+        """Return the mac address of the device."""
+        return self.entity_description.mac_address_fn(self._client_details)
+
+    @property
+    def hostname(self) -> str | None:
+        """Return hostname of the device."""
+        return self.entity_description.hostname_fn(self._client_details)
+
+    @property
+    def is_connected(self) -> bool:
+        """Return true if the device is connected to the network."""
+        return self.entity_description.is_connected_fn(self._client_details)
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return the unique id of the device."""
+        return f"scanner_{self._site_id}_{self._client_id}"
+
+
+class OmadaClientTrackerEntity(OmadaBaseTrackerEntity, TrackerEntity):
+    """Entity for a client connected to the Omada network."""
+
+    @property
+    def location_name(self) -> str | None:
+        """Return a location name for the current location of the device."""
+        return self.entity_description.location_fn(self._client_details)
```

Comment:
I felt like it kind of fitted because GPS coordinates were optional. But I'll remove it. I can't see another similar entity type that might make sense for the nearest AP. Perhaps if scanners had a way of changing areas?

Reply:
You can open a discussion and get ideas. Let's just remove this for now.